### ISSUE TITLE
cannot join any chat

### ISSUE THREAD

**Issue by wizdude** [nextcloud/talk-desktop#148]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

## Steps to reproduce
1. login to talk-desktop
2. click on any conversation (including the default "Talk Updates")
3. screen flashes quickly to the chat, but then returns back to "The conversation does not exist"

### Expected behaviour
I should be able to enter a chat as expected.

### Actual behaviour
You see the latest chat quickly flash up, then it returns with a list of chats on the left hand side
and "The conversation does not exist" on the right hand side.

## Desktop client

**Talk Desktop client version:** v0.3.0

**Operating system:** Windows

**Operating system version:** Windows 11

**Microphone available:** no

**Camera available:** no

## Server

**Nextcloud version:** 26.0.0.11

**Talk app version:** 16.0.2

**Custom Signaling server configured:** no

**Custom TURN server configured:** no

**Custom STUN server configured:** no

## Logs
no events are logged in nextcloud.log. let me know if i need to enable any specific debug
anywhere to produce logs.

### Client log
does talk-desktop have logs? i couldn't find them

<details>

```
Insert your browser log here, this could for example include:
a) The javascript console log
b) The network log
c) ...
```

</details>

**Comment by wizdude** [comment]
everything here is a fresh install. i did a fresh install of Nextcloud 26 in my lab and this is the first time i am trying the talk-desktop.

talk works fine in the web client. i have a test setup with one user on desktop and the other user on the mobile iOS client.
i can see chat updates happening in the left hand column of talk-desktop, but i cannot enter into the chat itself.
i do see the updated chat flash up briefly on the right hand side before it returns to the original "The conversation does not exist" screen on the right hand side.

**Comment by wizdude** [comment]
i spent a fair amount of time today working on this. i built a dev enviroment under WSL and cloned the current repository and did a fresh build for Windows.

this version works fine.

so either something in the last 21 commits has fixed the problem, or there was an issue with the release builds for 0.3.0.

i see two other issues #149 and #150 which are linux based and wonder if a new build on the current tree might fix their problems as well since the underlying issue looks pretty much identical.

side dev note - it took me half of today to work out what version of nodejs/npm to use to build this, since i was constantly experiencing build issues. i ended up having to install a full nextcloud docker dev environment just to work this out since i couldn't find it documented anywhere at all.

FWIW, the current Nextcloud docker dev environment is running node v16.20.0 with npm 8.19.4. i setup my build environment for this testing today under WSL running Ubuntu 22.04 and when i targeted v16.x with the nodesource packages it provided me with node v16.19.1 with npm 8.19.3 which i decided was close enough.

**Comment by nickvergessen** [comment]
> side dev note - it took me half of today to work out what version of nodejs/npm to use to build this, since i was constantly experiencing build issues. i ended up having to install a full nextcloud docker dev environment just to work this out since i couldn't find it documented anywhere at all.


https://github.com/nextcloud/talk-desktop/blob/main/package.json#L83-L86

I always dislike duplicating information, but maybe we can reference to check that file

**Comment by wizdude** [comment]
> https://github.com/nextcloud/talk-desktop/blob/main/package.json#L83-L86
> 
> I always dislike duplicating information, but maybe we can reference to check that file

my first build attempt was using node v18, but this failed. i kept getting dependency issues with vue-loader. i tried to add both version 15 and 16 into the build using "npm i vue-loader --save-dev" and similar but i ended up going down a deep rabbit hole. this was under WSL with Ubuntu 20.04. i decided to start again with WSL with Ubuntu 22.04 and wanted to match what the Nextcloud development environment was using and then ended up on node v16 and a successful build.

i spend ages trying to find what the "recommended environment" was for doing development work, but nothing was documented. i ended up spinning up a nextcloud docker dev environment and then started looking inside it for what versions were used.

so i've worked out that WSL Ubuntu 20.04 with node v18 doesn't work, and WSL Ubuntu 22.04 with node v16 does work.

i was just commenting on this, because the package.json indicated that v18 should have been ok, but perhaps it wasn't suitable for the OS version I was on.

**Comment by wizdude** [comment]
> i see two other issues #149 and #150 which are linux based and wonder if a new build on the current tree might fix their problems as well since the underlying issue looks pretty much identical.

just a quick update on this as well - i setup X under WSL so i could do some Linux builds and test them. it was successful with the current tree and i can chat successfully.

**Comment by pschopen** [comment]
I have access to two NC servers now both running Hub 4. One (a large instance) has the same problem, the other one (private, only one user), doesn't. I'm not the admin of the large installation, so I cannot tell any version numbers or so. But maybe ist still helps finding the issue?

**Comment by pschopen** [comment]
Apparently the App just fully crashes when opening a chat. The developer tools reveal an empty header and an empty body.

**Comment by pschopen** [comment]
Fixed in: https://github.com/nextcloud-releases/talk-desktop/releases/tag/v0.3.1

**Comment by ShGKme** [comment]
Thank you for your bug report. I'm closing it as duplication #138, which was originally in #73
Fixed in #141 and released in v0.3.1

> side dev note - it took me half of today to work out what version of nodejs/npm to use to build this, since i was constantly experiencing build issues. i ended up having to install a full nextcloud docker dev environment just to work this out since i couldn't find it documented anywhere at all.
> 
> FWIW, the current Nextcloud docker dev environment is running node v16.20.0 with npm 8.19.4. i setup my build environment for this testing today under WSL running Ubuntu 22.04 and when i targeted v16.x with the nodesource packages it provided me with node v16.19.1 with npm 8.19.3 which i decided was close enough.

@wizdude There are no special requirements: the current Node LTS (v18.16.0) works. The previous Node version also works. The same with compatible npm versions. Also, if your versions are incompatible, you will receive a warning from npm.

[nextcloud-docker-dev](https://github.com/juliushaertl/nextcloud-docker-dev/) has no containers for building either Talk Desktop or any frontend assets. The server containers have Node and npm, but they are for building the server assets as a part of the server container. You can use this environment for Nextcloud server development or hosting the dev server. But to develop other Nextcloud apps or desktop/mobile clients you may need to setup also a development environment according to the app requirements.

> i spend ages trying to find what the "recommended environment" was for doing development work, but nothing was documented. i ended up spinning up a nextcloud docker dev environment and then started looking inside it for what versions were used.
>
> so i've worked out that WSL Ubuntu 20.04 with node v18 doesn't work, and WSL Ubuntu 22.04 with node v16 does work.
>
> i was just commenting on this, because the package.json indicated that v18 should have been ok, but perhaps it wasn't suitable for the OS version I was on.

I tried your versions:
- Node v16.19.1
- npm v8.19.3
- Ubuntu 20.04.5 LTS
- WSL2 v1.1.6.0 on Windows 11

And it works fine without any issues. 

![image](https://user-images.githubusercontent.com/25978914/232330391-ba177483-b431-4053-80b5-532778f1ba91.png)
